### PR TITLE
Require Node.js 12 and move to ESM

### PR DESCRIPTION
**Pull request by lgrammel**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,9 +34,20 @@ function isTypeSync(fsStatType, statsMethodName, filePath) {
...
+const pathType = {
+	isFile,
+	isDirectory,
+	isSymlink,
+	isFileSync,
+	isDirectorySync,
+	isSymlinkSync
+};
+
+export default pathType;
```
Comment (sindresorhus) [inline index.js]: There should be no default export

File: package.json [hunk 2]
```diff
@@ -9,11 +9,13 @@
 		"email": "sindresorhus@gmail.com",
 		"url": "sindresorhus.com"
 	},
+	"type": "module",
+	"exports": "./index.js",
 	"engines": {
-		"node": ">=8"
+	  "node": ">=12"
```
Comment (sindresorhus) [inline package.json]: Incorrect indentation

File: test/eacces.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import fs from 'fs';
 import test from 'ava';
-import pathType from '../index.js';
+import * as pathType from '../index.js';
```
Comment (sindresorhus) [inline test/eacces.js]: This should import named exports.

**Comment by sindresorhus** [comment]
One thing I missed: You didn't update the readme usage example.

**Comment by lgrammel** [comment]
Oh - sorry about that, thanks for taking care of it!

**Pull request merged**